tests/api/python/tests.py.in

```diff
@@ -263,6 +264,77 @@ class TestSTP(unittest.TestCase):
             s.check(c == op(a))
             self.assertEqual(s.model()['c'], op(A) % 2**32)
 
+    def test_minisat(self):
+        """
+        Checks that the underlying solver can be set to be minisat
+        """
+
+        #
+        # Setting minisat should never fail
+        #
+        self.assertTrue(self.s.setMinisat())
+
+        #
+        # After calling set, the solver should be minisat
+        #
+        self.assertTrue(self.s.usingMinisat())
+
+    def test_simplifying_minisat(self):
+        """
+        Checks that the underlying solver can be set to be simplifying minisat
+        """
+
+        #
+        # Setting simplifying minisat should never fail
+        #
+        self.assertTrue(self.s.setSimplifyingMinisat())
+
+        #
+        # After calling set, the solver should be simplifying minisat
+        #
+        self.assertTrue(self.s.usingSimplifyingMinisat())
+
+    def test_cryptominisat(self):
+        """
+        Checks that the underlying solver can be set to be cryptominisat
+        """
+
+        if self.s.supportsCryptoMinisat():
+            #
+            # Setting cryptominisat should never fail, if STP is built to support it
+            #
+            self.assertTrue(self.s.setCryptoMinisat())
+
+            #
+            # After calling set, the solver should be cryptominisat
+            #
+            self.assertTrue(self.s.usingCryptoMinisat())
+        else:
+            #
+            # If STP doesn't support cryptominisat, then setting should rerturn false
+            #
+            self.assertFalse(self.s.setCryptoMinisat())
+
+    def test_riss(self):
```

Comment:
These tests have a lot of code duplication. Could they be turned into a parameterized test that does some dynamic method lookup with `getattr`?

Reply:
Yep!